pipelines/pipeline_katello_36_to_37_upgrade.yml

```diff
@@ -0,0 +1,107 @@
+- hosts: localhost
+  vars:
+    forklift_name: pipeline-katello-3.6-3.7
+    forklift_boxes:
+      pipeline-katello-3.6-3.7-centos7:
+        box: centos7
+        memory: 6144
+      pipeline-proxy-3.6-3.7-centos7:
+        box: centos7
+        memory: 3072
+  roles:
+    - forklift
+
+- hosts:
+  - pipeline-katello-3.6-3.7-centos7
+  - pipeline-proxy-3.6-3.7-centos7
+  become: yes
+  vars:
+    puppet_repositories_version: 5555
+    katello_repositories_version: 3.6
+    foreman_repositories_version: 1.17
+    foreman_installer_skip_installer: true
+  roles:
+    - umask
+    - selinux
+    - etc_hosts
+    - epel_repositories
+    - puppet_repositories
+    - foreman_repositories
+    - katello_repositories
+    - update_os_packages
+    - haveged
+    - disable_firewall
+
+- hosts: pipeline-katello-3.6-3.7-centos7
+  become: yes
+  vars:
+    foreman_installer_scenario: katello
+    foreman_installer_options_internal_use_only:
+      - "--foreman-admin-password {{ foreman_installer_admin_password }}"
+      - "--disable-system-checks"
+    foreman_installer_additional_packages:
+      - katello
+  roles:
+    - foreman_installer
+
+- hosts: pipeline-proxy-3.6-3.7-centos7
+  become: yes
+  vars:
+    foreman_proxy_content_server: pipeline-katello-3.6-3.7-centos7
+    foreman_installer_scenario: foreman-proxy-content
+    foreman_installer_options_internal_use_only:
+      - "--disable-system-checks"
+      - "--foreman-proxy-trusted-hosts {{ server_fqdn.stdout }}"
+      - "--foreman-proxy-trusted-hosts {{ ansible_nodename }}"
+      - "--foreman-proxy-foreman-base-url https://{{ server_fqdn.stdout }}"
+      - "--foreman-proxy-register-in-foreman true"
+      - "--foreman-proxy-oauth-consumer-key {{ oauth_consumer_key.stdout }}"
+      - "--foreman-proxy-oauth-consumer-secret {{ oauth_consumer_secret.stdout }}"
+      - "--foreman-proxy-content-certs-tar {{ foreman_proxy_content_certs_tar }}"
+      - "--foreman-proxy-content-parent-fqdn {{ server_fqdn.stdout }}"
+      - "--puppet-server-foreman-url https://{{ server_fqdn.stdout }}"
+      - "--foreman-proxy-content-pulp-oauth-secret {{ pulp_oauth_secret.stdout }}"
+    foreman_installer_additional_packages:
+      - foreman-installer-katello
+  roles:
+    - foreman_proxy_content
+    - foreman_installer
+
+- hosts: pipeline-katello-3.6-3.7-centos7
+  become: true
+  vars:
+    bats_tests:
+      - fb-test-katello.bats
+      - fb-content-katello.bats
+  roles:
+    - bats
+
+- hosts: pipeline-katello-3.6-3.7-centos7
+  become: true
+  vars:
+    katello_repositories_version: 3.7
+    foreman_repositories_version: 1.18
+    foreman_installer_upgrade: True
+    foreman_repositories_environment: staging
+    katello_repositories_environment: staging
+    foreman_installer_scenario: katello
+    foreman_installer_options_internal_use_only:
+      - "--upgrade-puppet"
```

Comment:
Looks like you're upgrading puppet? So you might want to try to explicitly test both Katello versions with both Puppet versions, similar to the [3.5 upgrade pipeline](https://github.com/theforeman/forklift/pull/710/files) (not yet merged). Just note that --upgrade-puppet is broken in 3.5, but it should work for you.

Reply:
I thought we removed `--upgrade-puppet` in 3.7 and included it in `--upgrade`

Reply:
Thanks. I removed `--upgrade-puppet`

Reply:
@ekohl Is there now no way to upgrade Katello but stay on P4?

Reply:
Manage your yum repositories so only P4 RPMs are available or pin them otherwise. We've modified `--upgrade` to just ensure we work with the currently installed Puppet version.